### PR TITLE
Fix default overload

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala [hunk 1]
```diff
@@ -667,8 +665,14 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
...
 
             if (isRawJSType(trgSym.toTypeConstructor)) {
-              assert(isScalaJSDefinedJSClass(defaultGetter.owner))
-              genApplyJSClassMethod(trgTree, defaultGetter, defaultGetterArgs)
+              if (isScalaJSDefinedJSClass(defaultGetter.owner)) {
+                genApplyJSClassMethod(trgTree, defaultGetter, defaultGetterArgs)
+              } else {
+                reporter.error(param.pos, "You cannot override a method of a " +
+                    "js.native class/trait with a default argument without a " +
+                    "default argument.")
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala]: The "with default argument without default argument part" is confusing. Consider a positive phrasing instead,  like

    When overriding a native method with default arguments, the overriding method must explicitly repeat the default arguments.

**Comment by sjrd** [comment]
Do we already have a pos test that when the default parameter is indeed explicitly specified, then it works?

**Comment by sjrd** [comment]
That's all.

**Comment by gzm0** [comment]
Updated.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**